### ISSUE TITLE
Rule proposal: allow kebab-case

### ISSUE THREAD

**Issue by gregordotjs** [typescript-eslint/typescript-eslint#4667]
### Description

AFAIK kebab-case is currently not supported, and there are some edge cases in Object Literal Property name where this is necessary (see example below). The workaround is to disable format for Object Literal Property, but this could be solved by allowing kebab-case.

### Fail

```ts
const commonlyUsedParams: LabelHTMLAttributes<HTMLLabelElement> = {
    id: 'some-id',
    className: 'some-className',
    tabIndex: -1,
    ariaLabel: 'some-aria-label',
};
```

Error: "Object literal may only specify known properties, but 'ariaLabel' does not exist in type 'LabelHTMLAttributes<HTMLLabelElement>'."

### Pass

```ts
const commonlyUsedParams: LabelHTMLAttributes<HTMLLabelElement> = {
    id: 'some-id',
    className: 'some-className',
    tabIndex: -1,
    'aria-label': 'some-aria-label',
};
```
Currently there's an error: "Object Literal Property name 'aria-label' must match one of the following formats: camelCase, snake_case, UPPER_CASE, PascalCase"

**Comment by bradzacher** [comment]
I assume you're talking about the `naming-convention` rule?

You can add your own support for kebab case using the `custom` config.

https://typescript-eslint.io/rules/naming-convention#custom

You can also specifically match quoted names using the `requiresQuotes` modifier.

Kebab case isn't something that's commonly used across JS because it requires string literals - so I don't think it's worth adding it as a pre-built format option.

**Comment by Retro64** [comment]
@bradzacher  I know this issue is closed, but according to https://typescript-eslint.io/rules/naming-convention/#how-does-the-rule-evaluate-a-names-format it looks for me `custom` is checked before format and therefore something like `['snake_case', 'kebab-case']` cannot be designed with `custom` in an easy way. Either the Regex need to contain all formats including kebab-case, making format kind of obsolete or - if you define kebab as Regex - nothing will be validated, as all other cases break the Regex. Defining one seperator twice - once with format, once with the kebab-case regex - also did not work out. It might be a rare case, but e.g. designing http headers as object might need kebab-case - even in JS. 

Not sure if my team simply writes uncommon code, but when updating the rules to a version not accepting kebab-case as camelCase (which for sure was a bug), we ended up with quite many ignores throughout the code. But currently after playing around and some try and error, at least do not see any possibility to design something working like `format: ['snake_case', 'kebab-case', 'camelCase', 'PascalCase']` with `custom` - except for building a big and complex Regex covering all accepted formats...

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/naming-convention/#ignore-properties-that-require-quotes

You can build upon this to specifically target your quoted kebab case names.
